### DISCUSSION TITLE
OLED 2.42 with SSD1309 SPI:   5V or 3.3V?   NONAME0 or NONAME2?   SDA, CLK, ... only 3.3V?

### DISCUSSION THREAD

**Discussion by igel1** [olikraus/u8g2#2417] [General]
Hi folks,

I bought an OLED 2.42 with SSD1309 SPI and now I am quite unsecure how to deal with it.
I would like to use it on my Arduino UNO, Nano r3 and 2560 Rev3 and later on with STM32 as well 
(Arduinos run on 5V, the STM boards runs on 3.3V).

I bought this OLED:   https://www.aliexpress.com/item/1005006473260235.html#nav-specification
(you can find the schematics on that page and also in the attachment of this post)

1. Question:    As far as I can see from the schematics, this board has an xc6206 voltage regulator,
                       which probably outputs 3.3V to the SSD1309.
                       So I guess, I can power the VDD pin with something in between 3.6V ... 6V - correct?
                       
2. Question:   As far as I understand from the datasheet, the SSD1309 runs at 3.3V
                       So I do have to use level shifters  (5V -> 3.3V) for the SDA, SCK, DC, CS, RES - correct?
                       (at least for my Arduinos,  not for the STM32)

3. Question:    When do I have to use the "... _NONAME0_..." and when the "..._NONAME2_..." version of the constructor?
                        And when do I use the "... _2ND_ ..." or the version without enumeration in the constructor's name?

4. Question:   What do you think: does this wiring & all of these 3 constructors work for the above mentioned OLED?

```
           Wiring:
            OLED 2.42´´ SPI - SSD1309         UNO                 Nano

            CS    (chip select)               10 (SS)              12
            DC    (data commands)              9                   11
            RES   (reset)                      8                   13
            SDA   (serial data)               11 (MOSI)             9       
            SCLK  (clock)                      3 (SCK)             10
            VDD   (supply voltage)             +5 V                +5 V
            VSS   (GND - ground)               GND                 GND
          
          
//U8G2_SSD1309_128X64_NONAME0_1_4W_HW_SPI(rotation,      cs,   dc [, reset]) [page buffer, size = 128 bytes]
//U8G2_SSD1309_128X64_NONAME2_1_4W_HW_SPI u8g2(U8G2_R0,   10,     9,    8);      // For Arduino Uno:
//U8G2_SSD1309_128X64_NONAME0_1_4W_HW_SPI u8g2(U8G2_R0,   10,     9,    8);      // For Arduino Uno:
//U8G2_SSD1309_128X64_NONAME0_1_4W_HW_SPI u8g2(U8G2_R0,   12,    11,   13);     // For Arduino Nano
```

Sorry for asking so many question - I am slightly lost at the moment (and I have only one display - no spare parts ...)

Kind regards

A.S.

![Schema-OLED-2 42-SSD1309-SPI-Vers-4 2](https://github.com/olikraus/u8g2/assets/14350970/459ad084-bd40-4c02-bd4a-d7acf94b47eb)

![OLED-2 42-SSD1309-SPI-Vers-4 2](https://github.com/olikraus/u8g2/assets/14350970/8b7b8bf2-79e8-4c29-adaa-17aca73f6182)

**Comment by olikraus** [comment]
Q1: Hmm difficult to answer and also the schematic is not so clear to me.
Q2: yes
Q3: Needs to be tested. At least there is no danger to your display if you use the wrong constructor. 
Q4: HW SPI requires the lines marked for clock and data on the uno and nano boards. I think for uno it is 13 for the clock.
Avoid using the MISO pin with AVR.

**Comment by igel1** [comment]
Thanks for your help, Oli !

Meanwhile I solved all questions:

Q1: Yes
Q2: Yes
Q3: This worked for me:   
```
U8G2_SSD1309_128X64_NONAME0_F_4W_HW_SPI u8g2(U8G2_R0, 10, 9, 7);
```
Q4: The UNO and the Nano have the same wiring for SSD1309 controlled displays.
       This worked for me:
       
```           
      Wiring:
            OLED 2.42´´ SPI - SSD1309         UNO                 Nano
           ------------------------------------------------------------
            CS    (chip select)               10 (SS)             10
            DC    (data commands)              9                   9
            RES   (reset)                      7                   7
            SDA   (serial data)               11 (MOSI)           11       
            SCLK  (clock)                     13 (SCK)            13
            VDD   (supply voltage)            +5 V                +5 V
            VSS   (GND - ground)              GND                 GND
```

Kind regards

A.S.
